src/mlpack/core/data/load.hpp

```diff
@@ -283,59 +208,13 @@ bool Load(const std::string& filename,
  * @param transpose If true, transpose the matrix after loading.
  * @return Boolean value indicating success or failure of load.
  */
-template<typename eT, typename PolicyType>
+template<typename MatType, typename eT, typename PolicyType>
 bool Load(const std::string& filename,
           arma::Mat<eT>& matrix,
           DatasetMapper<PolicyType>& info,
           const bool fatal = false,
           const bool transpose = true);
 
```

Comment:
I have a small doubt here.
```
template<typename MatType, typename eT>
bool Load()
```
Now when I have to call the `Load` we will use
`Load<int, arma::Mat<int>>`
But I don't want to set the Mat's datatype here as we have eT for that.
Is there a workaround for this?

Reply:
In general the template parameters can be automatically deduced.  So a user doesn't need to specify them.  This will work:

```
arma::mat matrix;
data::DatasetInfo di;
data::Load("file.sv", matrix, di);
```

I haven't looked at any of the rest of the PR, but my guess is that you don't need `eT` anymore and you should just use `MatType& matrix` as a parameter. :+1:

Reply:
Yea, that's what I was thinking, we can remove `eT`, and have only `MatType` as template parameter.